scripts/release/group_test.py

```diff
@@ -78,67 +92,68 @@ def run_pytest_jobs(pytest_jobs: dict, my_env: dict):
     test_coverage = {}
     covered_assets = []
 
-    if group_pre:
-        check_call(f"python {group_pre}", env=my_env, shell=True)
-
-    pytest_jobs = {}  # pytest job path -> assets coverage dict
-    with open(tests_dir / TEST_YML) as fp:
-        data = yaml.load(fp, Loader=yaml.FullLoader)
-        for job, job_data in data[test_group]['jobs'].items():
-            if 'pytest_job' in job_data:
-                pytest_jobs[tests_dir / job_data['pytest_job']] = job_data['assets']
+    with set_directory(tests_dir):
+        if group_pre:
+            check_call(f"python {group_pre}", env=my_env, shell=True)
+
+        pytest_jobs = {}  # pytest job path -> assets coverage dict
+        with open(tests_dir / TEST_YML) as fp:
+            data = yaml.load(fp, Loader=yaml.FullLoader)
+            for job, job_data in data[test_group]['jobs'].items():
+                if 'pytest_job' in job_data:
+                    pytest_jobs[tests_dir / job_data['pytest_job']] = job_data['assets']
+                else:
+                    if 'pre' in job_data:
+                        logger.print(f"Running pre script for {job}")
+                        proc = check_call(f"python3 {tests_dir / job_data['pre']}", env=my_env, shell=True)
+                    print(f'Loading test job {job}')
+                    try:
+                        test_job = load_job(tests_dir / job_data['job'])
+                        logger.print(f'Running test job {job}')
+                        test_job = ml_client.jobs.create_or_update(test_job)
+                    except Exception as ex:
+                        logger.log_warning(
+                            f"catch error submitting {job_data['job']} with exception {ex}")
+                        failed_jobs.append(job)
+                        continue
+                    test_coverage[test_job] = job_data.get("assets", [])
+                    logger.print(f'Submitted test job {job}')
```

Comment:
Please prefer double quotes for all strings except for dictionary keys, which should be single quotes.

Reply:
Done